pkgs/applications/version-management/git-and-tools/git/default.nix

```diff
@@ -76,6 +77,9 @@ stdenv.mkDerivation {
   configureFlags = stdenv.lib.optionals (stdenv.buildPlatform != stdenv.hostPlatform) [
     "ac_cv_fread_reads_directories=yes"
     "ac_cv_snprintf_returns_bogus=no"
+    # Provide curl manually, do not rely on `curl-config`
+    # See also `CURL_LDFLAGS` below.
+    "--with-curl=${symlinkJoin { name = "curl-libs-and-headers"; paths = [ curl.out curl.dev ]; }}"
```

Comment:
Actually maybe try `CURL_CONFIG=${curl.dev}/bin/curl-config`? from here: https://github.com/git/git/blob/master/configure.ac#L597

Reply:
Using symlinkJoin is usually a bad idea

Reply:
> Using symlinkJoin is usually a bad idea

@matthewbauer Why?

Reply:
It breaks the closure size reductions from multiple outputs. When you link to that lib directory, you will also get the headers in closures.

Reply:
Ah, I see. That makes sense.

What should I do though, given that git has only a single flag as opposed to one for headers and libs each?

I can try the `curl-config` approach, but using explicit flags sems to have less surprises so that might be preferable.

Do you no longer have the concerns about `curl-config` from #61527, or is the difference of your proposal above that we'd pass the `CURL_CONFIG` manually and that would somehow be better / address your previous concern of

> Otherwise you wind up linking to the nonstatic curl I think

?

Reply:
@matthewbauer A short ping for my questions above :)

Reply:
Sorry. Yeah you can use curl-config, you just can't put it in nativeBuildInputs, otherwise you get the build system's curl-config binary, not the targeted system. I believe putting ```"CURL_CONFIG=${curl.dev}/bin/curl-config"``` in configureFlags should be okay (if a little awkward).

Reply:
@matthewbauer Except that curl-config won't work if you're cross-compiling to a different architecture, e.g. armv7l.

Reply:
@kisik21 Some packages natively compile the config tools that go in the `dev` output precisely for this reason. It's ugly either way, but doing that is practical.